src/main/java/cn/nukkit/level/manager/FinishingTask.java

```diff
@@ -0,0 +1,53 @@
+package cn.nukkit.level.manager;
+
+import cn.nukkit.level.chunk.Chunk;
+import cn.nukkit.level.chunk.IChunk;
+import cn.nukkit.level.chunk.LockableChunk;
+import cn.nukkit.level.generator.Generator;
+import com.google.common.base.Preconditions;
+import lombok.AccessLevel;
+import lombok.NoArgsConstructor;
+import lombok.NonNull;
+import net.daporkchop.lib.random.PRandom;
+import net.daporkchop.lib.random.impl.FastPRandom;
+
+import java.util.List;
+import java.util.concurrent.locks.Lock;
+import java.util.function.BiFunction;
+
+/**
+ * Delegates chunk finishing to a {@link Generator}.
+ *
+ * @author DaPorkchop_
+ */
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+public final class FinishingTask implements BiFunction<Chunk, List<Chunk>, Chunk> {
+    public static final FinishingTask INSTANCE = new FinishingTask();
+
+    @Override
+    public Chunk apply(@NonNull Chunk chunk, List<Chunk> chunks) {
+        if (chunk.isFinished())    {
```

Comment:
reformat

Reply:
what exactly is wrong here? i mean there are some extra spaces at the end, but i'm just using my default code formatting config. it's not like there's even a strictly defined code style at all (#1093)